pkg/api/datasources.go

```diff
@@ -336,16 +337,18 @@ func validateURL(cmdType string, url string) response.Response {
 // This is done to prevent data source proxy from being used to circumvent auth proxy.
 // For more context take a look at CVE-2022-35957
 func validateJSONData(ctx context.Context, jsonData *simplejson.Json, cfg *setting.Cfg, features *featuremgmt.FeatureManager) error {
-	if jsonData == nil || !cfg.AuthProxyEnabled {
+	if jsonData == nil {
 		return nil
 	}
 
-	for key, value := range jsonData.MustMap() {
-		if strings.HasPrefix(key, datasources.CustomHeaderName) {
-			header := fmt.Sprint(value)
-			if http.CanonicalHeaderKey(header) == http.CanonicalHeaderKey(cfg.AuthProxyHeaderName) {
-				datasourcesLogger.Error("Forbidden to add a data source header with a name equal to auth proxy header name", "headerName", key)
-				return errors.New("validation error, invalid header name specified")
+	if cfg.AuthProxyEnabled {
```

Comment:
```suggestion
	if cfg.AuthProxyEnabled {
```

Same way we do here, to check that the AuthProxyEnabled, we should check that cfg.BasicAuthEnabled. Because validation of the json

Reply:
Yes, makes sense I think

Reply:
Hm, we can skip it since we're checking this in a middleware. The purpose of this function is only validation of the params, so let's skip it. Otherwise we need to parse data source instance settings first which is no here.